code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm

```diff
@@ -81,14 +92,21 @@
 
 
 /obj/machinery/atmospherics/components/binary/thermomachine/update_icon_state()
-	if(cooling)
-		icon_state_off = "freezer"
-		icon_state_on = "freezer_1"
-		icon_state_open = "freezer-o"
+	if(target_temperature <= BODYTEMP_COLD_WARNING_3)
+		set_greyscale(colors=list(thermo_colors["purple"]))
+	else if(target_temperature <= BODYTEMP_COLD_WARNING_2)
+		set_greyscale(colors=list(thermo_colors["blue"]))
+	else if(target_temperature <= BODYTEMP_COLD_WARNING_1)
+		set_greyscale(colors=list(thermo_colors["cyan"]))
+	else if(target_temperature <= BODYTEMP_HEAT_WARNING_1)
+		set_greyscale(colors=list(thermo_colors["green"]))
+	else if(target_temperature <= BODYTEMP_HEAT_WARNING_2)
+		set_greyscale(colors=list(thermo_colors["yellow"]))
+	else if(target_temperature <= BODYTEMP_HEAT_WARNING_3)
+		set_greyscale(colors=list(thermo_colors["orange"]))
 	else
-		icon_state_off = "heater"
-		icon_state_on = "heater_1"
-		icon_state_open = "heater-o"
+		set_greyscale(colors=list(thermo_colors["red"]))
+
```

Comment:
Maybe it would be better to do this with a var and have only one case of set_greyscale at the end? might make things a bit cleaner

Reply:
Agreed.  Fixed with ec061af06dd50d4e290e1f9ce46f5485f251beba